PowerShell Contributing.md

```diff
@@ -0,0 +1,479 @@
+VSCode will connect to that container and you can work with it from there.
+
+You will need:
+
+- [Docker](https://www.docker.com/products/docker-desktop), configure a shared folder
+- Give the Docker VM more RAM than just the default 2048MB if you can spare it, it will give you a nicer experience.
+- VSCode [remote-development extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
+
+Read more [here](https://code.visualstudio.com/docs/remote/remote-overview).
+
+The first time, it will ask you to install recommended VSCode extensions, and loading the project may fail because of the missing extensions. 
+
+- Install the recommended extensions and reload
+- If VSCode tells you 'There are unresolved dependencies.' Click `Restore`
+
+Now your environment should be all set.
+
+- The terminal defaults to Bash. To start PowerShell, simply type `pwsh` 
+- You can now load the `build.psm1` module with `Import-Module ./build.psm1` 
+- In some cases you may need to run `Start-PSBuild -Clean`
+
+### How to run a test
+
+To run a test, you need to build the pwsh binaries first, and you need the test-tool.
+Also note, if you run all tests, that's going to take some time.
+Good thing there is an easy way!
+
+In the base of the repository there is a module 'build.ps1', that has all kind of handy Cmdlets in it for this kind of thing.
+Load it like this:
+
+`Import-Module .\build.psm1`
+
+(If you want to know which tools are inside, use `Get-Command -Module build`)
+
+The very fist time, you need to set up a few things.
+To do that, run `Start-PsBootstrap` (may need sudo/adminstrative privileges, depending on your system.
+On a docker container, it needs the `-NoSudo` switch).
+
+You also need to build the pwsh binaries for the tests to run against: `Start-PsBuild`
+
+> Tip: If you encounter weird build errors, just run `Start-PSBuild -Clean`.
+This is generally a good idea every time you update your branch, or after a rebase.
+
+Now you could simply run `Start-PSPester` to kick off a build, and run ALL Pester tests.
+This would take some time, and this is still not what you want to do when writing extra tests, or improving existing ones.
+So what do we do instead? We tell `Start-PSPester` which testfile we want to run:
+
+`Start-PSPester -Path .\test\powershell\Modules\Microsoft.PowerShell.Core\Get-Command.Tests.ps1`
+
+You will notice that it rebuilds test tool binaries each time you use Start-PSPester.
+This takes time and usually we only need to build these binaries once.
+To suppress new builds, you add the `-SkipTestToolBuild` switch:
+
+`Start-PSPester -Path .\test\powershell\Modules\Microsoft.PowerShell.Core\Get-Command.Tests.ps1 -SkipTestToolBuild`
+
+When writing and debugging your tests, remember that you can set breakpoints and step through your Pester code, this can help you find out where things are not going as they | Should -Be :-).
```

Comment:
```suggestion
When writing and debugging your tests, remember that you can set breakpoints and step through your Pester code, this can help you find out where things are not going as they `| Should -Be` :-).
```